llvm/include/llvm-c/CAS/PluginAPI_types.h

```diff
@@ -0,0 +1,85 @@
+/*===-- llvm-c/CAS/PluginAPI_Types.h - CAS Plugin Types Interface -*- C -*-===*\
+|*                                                                            *|
+|* Part of the LLVM Project, under the Apache License v2.0 with LLVM          *|
+|* Exceptions.                                                                *|
+|* See https://llvm.org/LICENSE.txt for license information.                  *|
+|* SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception                    *|
+|*                                                                            *|
+|*===----------------------------------------------------------------------===*|
+|*                                                                            *|
+|* The types for the LLVM CAS plugin API.                                     *|
+|*                                                                            *|
+\*===----------------------------------------------------------------------===*/
+
+#ifndef LLVM_C_CAS_PLUGINAPI_TYPES_H
+#define LLVM_C_CAS_PLUGINAPI_TYPES_H
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#define LLCASAPI_VERSION_MAJOR 0
+#define LLCASAPI_VERSION_MINOR 1
+
+typedef struct llcasapi_cas_options_s *llcasapi_cas_options_t;
+typedef struct llcasapi_cas_s *llcasapi_cas_t;
+
+/**
+ * Digest hash bytes.
+ */
+typedef struct {
+  const uint8_t *data;
+  size_t size;
+} llcasapi_digest_t;
+
+/**
+ * Data buffer for stored CAS objects.
+ */
+typedef struct {
+  const void *data;
+  size_t size;
+} llcasapi_data_t;
+
+/**
+ * Identifier for a CAS object.
+ */
+typedef struct {
```

Comment:
Do we need `objectid_t`? I think if we allow `ObjectRef` is a Trie that points to nothing, we can only expose `ObjectRef` without ever need object id? Maybe just to satisfy the `CASID` conversion?

Reply:
`objectid_t` is essentially the equivalent of `ObjectRef`, what do you mean "we can only expose `ObjectRef` without ever need object id?"?

Reply:
@cachemeifyoucan clarified offline it was just a misunderstanding related to `llcas_object_refs_t`.